felix/config/config_params.go

```diff
@@ -331,7 +331,9 @@ type Config struct {
 	// - calicoIPAM: use IPAM data to contruct routes.
 	RouteSource string `config:"oneof(WorkloadIPs,CalicoIPAM);CalicoIPAM"`
 
-	RouteTableRange idalloc.IndexRange `config:"route-table-range;1-250;die-on-fail"`
+	// RouteTableRange is deprecated in favour of RouteTableRanges
+	RouteTableRange  idalloc.IndexRange   `config:"route-table-range;1-250;die-on-fail"`
```

Comment:
Feel like one of these should have no default so that:
* if the user has already specified `RouteTableRange` in their config, we honour that.
* Otherwise, we use the new `RouteTableRanges` value

Reply:
In the case where both are set, presumably `RouteTableRanges` should take precedence, however, is there a way of differentiating between a default-set value and a user-set value?

The reason I ask is because if I remove the old arg's default in favour of an empty val...
 - I can always see when the old arg is explicitly-set, but
 - can only tell when the new arg is non-default.
 - if for some reason the user set the new arg explicitly to the same value as default, but also has the old arg set to something, it would be ambiguous which value I should run with. I _should_ be taking the new one, but it _looks_ like a default val...

Hope this is making sense... Im wondering should the _old_ arg be the one to take precedence, but also have no default. This would disambiguate the situation, but Im not sure if It's the right call.

Reply:
We should probably start moving to a world where defaults are done in the API and not in felix code.

Doing it in code means upgrades can break users more easily (because an implicit default changed). Doing it via the API means changing felix out won't change the behavior of the cluster.

I guess sometimes we want to be able to sneakily change something, but in general that's bad practice and imo would be better to be explicit about the config running on the system.

This is how we handle defaulting for all of the operator APIs, for example